### ISSUE TITLE
using memmap as a library in a plugin

### ISSUE THREAD

**Issue by rareguy** [volatilityfoundation/volatility3#917]
This is more of a question that came up when I was trying to make a plugin.

What I want: Using `memmap --dump` to either output as a file or a python bytestring, and then use it for another plugin which I'm trying to make.
What I've tried:
- Importing `memmap` and trying to use the run() method, then I realized it's not possible because it's not a `classmethod`
- Using `process_dump()` from `pslist` and then I realized it does not have the same result as `memmap --dump`
Can anyone assist me with this?

**Comment by rareguy** [comment]
I solved my problem. I took the code from the Volatility 3 plugin contest winner ForensicxLab [here](https://github.com/forensicxlab/volatility3_plugins/blob/ed9be16e0f7beac82369921b85f509d700eb51cb/anydesk.py#L42), which was also taken from `DumpFiles` plugin. It can receive a `FILE_OBJECT` file and returns a `bytestring` of the raw file byte and filename.

Though for development copy-pasting this bit of code is quite messy, I would suggest to add this as a `classmethod` for `DumpFiles` so that it would work for plugin developer to process any file extracted from the memory.

**Comment by ikelos** [comment]
A very similar question was asked on slack recently.  I recommended the following:

```
Hiya, I wouldn't recommend creating the file and then reading it, that's pretty inefficient.  You might as well use the guts of memmap (there's not a huge amount there) and do something like:

                proc_layer_name = proc.add_process_layer()
                proc_layer = self.context.layers[proc_layer_name]
                data = proc_layer.read(0x0, proc_layer.maximum_address, pad=True)

Then you've got the process layer and you can read/process just the bits you might need (or the whole lot, as in the last line)?
```

I don't think that amount of code is worth a whole extra method for such a specific need.  Ideally you'd be taking the process layer's memory and doing whatever you'd do to a byte string or file object contents.  I think there'd need to be good justification for splitting those few lines of code out into a separate method and copying the contents of a layer into memory and a big buffer just for the sake of not having it in a layer object?  What would this classmethod look like and what would it do differently from the couple of lines above?

**Comment by ikelos** [comment]
I'm marking this as needs more info to allow time for a response, but I'm not seeing the value of an additional method so I'll likely close this off further down the line without a response...

**Comment by rareguy** [comment]
Hey, thanks for the answer. Sorry for the weird wordings I made. I had spent too much time meddling with the library that the question was irrelevant from my own response (the second one). The answer was actually the whole another problem which I solved (and not related with the original question) regarding dumping files through a classmethod. But the suggestion still stands, kinda inefficient if I had to copy paste to get the entire filebyte.

Anyway, for the original question, is that code bit can be narrowed down for specific process pid?
